File: benchmarks/mandelbrot/mandelbrot.lua
local WIDTH = 450
local HEIGHT = 450
local MAXITER = 400

local checksum = 0
for py = 0, HEIGHT - 1 do
    local cy = (py / HEIGHT) * 3.0 - 1.5
    for px = 0, WIDTH - 1 do
        local cx = (px / WIDTH) * 3.0 - 2.0
        local zx = 0.0
        local zy = 0.0
        local iter = 0
        while (zx * zx + zy * zy <= 4.0) and (iter < MAXITER) do
            local new_zx = (zx * zx - zy * zy) + cx
            zy = (2.0 * zx * zy) + cy
            zx = new_zx
            iter = iter + 1
        end
        checksum = checksum + iter
    end
end
print(checksum)

File: benchmarks/mandelbrot/mandelbrot.js
const WIDTH = 450, HEIGHT = 450, MAXITER = 400;

let checksum = 0;
for (let py = 0; py < HEIGHT; py++) {
    const cy = (py / HEIGHT) * 3.0 - 1.5;
    for (let px = 0; px < WIDTH; px++) {
        const cx = (px / WIDTH) * 3.0 - 2.0;
        let zx = 0.0, zy = 0.0, iter = 0;
        while (zx * zx + zy * zy <= 4.0 && iter < MAXITER) {
            const new_zx = (zx * zx - zy * zy) + cx;
            zy = (2.0 * zx * zy) + cy;
            zx = new_zx;
            iter = iter + 1;
        }
        checksum = checksum + iter;
    }
}
console.log(checksum);
